src/imp/noop_imp.js

```diff
@@ -1,18 +1,25 @@
+function NoopSpanContextImp() {
+}
+NoopSpanContextImp.prototype.setBaggageItem = function (key, value) {
+}
+NoopSpanContextImp.prototype.getBaggageItem = function (key) {
+}
+
 function NoopSpanImp() {
 }
 NoopSpanImp.prototype.tracer = function () {
     return null;
 }
+NoopSpanImp.prototype.context = function () {
+    // XXX what's the deal here? Why can tracer() return null?
```

Comment:
`tracer()` can return `null` because this is returning the underlying imp object, not what the caller of the OpenTracing library sees. The caller still sees a OpenTracing `Tracer`. See https://github.com/opentracing/opentracing-javascript/blob/master/src/span.js#L25

Also, for what it's worth, this file is sorely missing a comment that implementation code here is used solely for the unit tests and will never be hit by in normal usage.